aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -705,6 +705,64 @@ struct ReductionMulOp {
   __forceinline__ scalar_t identity_cpu() const { return 1; }
 };
 
+void _apply_sparse_csr_linear_solve(
+  const Tensor& A,
+  const Tensor& b,
+  const bool left,
+  const Tensor& x) {
+#if defined(USE_ROCM) || !defined(USE_CUDSS)
+  TORCH_CHECK(
+      false,
+      "Calling linear solver with sparse tensors requires compiling ",
+      "PyTorch with CUDA cuDSS and is not supported in ROCm build.");
+#else
+  TORCH_CHECK(A.is_sparse_csr(), "A must be a CSR matrix");
+  TORCH_CHECK(b.dim() == 1, "b must be a 1D tensor");
+  TORCH_CHECK(b.stride(0) == 1, "b must be a column major tensor");
+  TORCH_CHECK(b.size(0) == A.size(-1), "linear system size mismatch.");
+  TORCH_CHECK(A.dtype() == b.dtype(), "A and b must have the same dtype");
+  TORCH_CHECK(left == true, "only left == true is supported by the Sparse CSR backend")
```

Comment:
Checks are needed to enforce the size/stride requirements on `x`. There should also be a guard against `x` or `b` having non-strided layout.

Reply:
fixed by https://github.com/pytorch/pytorch/pull/129856/commits/9c61f3dd0bd1d09bfafe198f0b6c0ae26a029050